compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2659,6 +2659,10 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         fullyInstantiated(tp2) && !tp1.classSymbols.exists(_.derivesFrom(tp2.symbol))
       case (tp1: TypeRef, tp2: TermRef) if isEnumValue(tp2) =>
         fullyInstantiated(tp1) && !tp2.classSymbols.exists(_.derivesFrom(tp1.symbol))
+      case (tp1: RefinedType, tp2: RefinedType) if tp1.refinedName == tp2.refinedName =>
+        provablyDisjoint(tp1.parent, tp2.parent) || provablyDisjoint(tp1.refinedInfo, tp2.refinedInfo)
+      case (tp1: TypeAlias, tp2: TypeAlias) =>
```

Comment:
The overall expression matches on `(tp1.dealias, tp2.dealias)`, does it make that case unreachable?

Reply:
No, dealias does not unwind refined types.

Reply:
I think there was a miscommunication: this (TypeAlias, TypeAlias) case seems redundant indeed.

Reply:
Wait, what?  Dealias doesn't dealias TypeAlias's?

Reply:
No, it dealiases TypeRefs whose inofo is a TypeAlias. A TypeAlias is a subtype of TypeBounds. It is information _about_ a type, not a type itself.